File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i,b,c=0;
        vector<int> o;
        for(i=0;i<nums.size();i++){
            //if(target>nums[i]){
                b = target - nums[i];
            //}
            
            for(int j=i+1;j<nums.size();j++){
                if(b==nums[j]){
                    o.push_back(i);
                    o.push_back(j);
                    return o;
                 }
            }
        }
        return o;
    }
    
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indexMap = new HashMap<>();
        int[] res = new int[2];

        for(int i = 0; i<nums.length; i++){
            int n = target - nums[i];
            if(indexMap.containsKey(n)){
                res[0] = indexMap.get(n);
                res[1] = i;
                break;
            }
            indexMap.put(nums[i], i);
        }
        return res;
    }
}